reviews/views.py

```diff
@@ -0,0 +1,92 @@
+import json, re, bcrypt, jwt
+
+from django.views         import View
+from django.http          import JsonResponse
+
+from reviews.models       import Review, ReviewImage
+from users.models         import User, WishList, Rating
+from users.utils          import login_decorator
+from restaurants.models   import Restaurant
+
+# Create your views here.
+class Review_View(View):
+    @login_decorator
+    def post(self, request, restaurant_id):
+        try:
+            data    = json.loads(request.body)
+            user    = request.user
+            reviews = Review.objects.filter(user_id = user.id)
+
+            for review in reviews:
+                if review.restaurant_id == restaurant_id:
+                    return JsonResponse({'MESSAGE':'EXISTS'}, status=400)
+
+            Review.objects.create(
+                restaurant_id = restaurant_id,
+                user_id       = User.objects.get(id=user.id).id,
+                description   = data['description'],
+            )
+
+            review_id = Review.objects.filter(user_id = user)
+            review_id = review_id[0].id
+
+            ReviewImage.objects.create(
+                review_id = review_id,
+                image     = data['image'],
+            )
+
+            Rating.objects.create(
+                user_id       = User.objects.get(id=user.id).id,
+                restaurant_id = restaurant_id,
+                rating        = data['rating']
+            )
+
+            return JsonResponse({'MESSAGE':'SUCCESS'},status = 201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+    @login_decorator 
+    def put(self, request, restaurant_id):
+        try:
+            data = json.loads(request.body)
+            user = request.user
+
+            Review.objects.filter(user_id = user.id, restaurant_id = restaurant_id).update(
+                    restaurant_id = restaurant_id,
+                    user_id       = User.objects.get(id=user.id).id,
+                    description   = data['description'],
+                    )
+            user_review = Review.objects.filter(user_id = user.id)
+            review_id = user_review[0].id
```

Comment:
```suggestion
            review = Review.objects.filter(user_id = user.id, restaurant_id = restaurant_id)
            review.update(
                    restaurant_id = restaurant_id,
                    user_id       = User.objects.get(id=user.id).id,
                    description   = data['description'],
                    )
```

이런식으로 작성할 경우, 아래에서도 filter 를 통한 반환값을 재 사용할 수 있습니다.
추가적으로 filter 한 후에 반환되는 객체가 있는지 검증해주세요. 그리고 user_review[0] 와 같이 인덱싱 할 경우, 해당하는 객체가 없을 때 어떤 에러가 발생하는지 확인해주세요.

Reply:
아... 반환값 재사용을 생각을 하지 못했습니다 ㅠㅠ 객체가 존재하지 않으면 작동하지 않는 다는 조건문을 추가하도록 하겠습니다.